packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -59,23 +60,34 @@ export function getProxyForField(field: FlexTreeField): TreeNode | TreeValue | u
 		case FieldKinds.required: {
 			const asValue = field as FlexTreeTypedField<FlexFieldSchema<typeof FieldKinds.required>>;
 
-			// TODO: Ideally, we would return leaves without first boxing them.  However, this is not
-			//       as simple as calling '.content' since this skips the node and returns the FieldNode's
-			//       inner field.
-			return getOrCreateNodeProxy(asValue.boxedContent);
+			const maybeUnboxedContent = asValue.content;
+			if (maybeUnboxedContent === undefined) {
+				return undefined;
+			}
+			if (isTreeValue(maybeUnboxedContent)) {
```

Comment:
Since you have nice benchmarks to measure things, you might want to try isTreeNode (the opposite check) instead since that might be faster (there are a bunch of leaf types which need special detection, especially handles and null)

Reply:
isFlexTreeNode was the one, but it does give better benchmark results (including tests that actually use flex nodes, not just unhydrated nodes), I'll put out the changes shortly.